File: compass.js
input.onButtonPressed(Button.A, function () {
    basic.clearScreen()
    printing = true
    basic.showNumber(input.compassHeading())
    printing = false
})
input.onButtonPressed(Button.AB, function () {
    callArrow()
    basic.pause(1000)
    basic.clearScreen()
})
input.onButtonPressed(Button.B, function () {
    if (!(ledOn)) {
        ledOn = true
    } else {
        ledOn = false
        if (!(printing)) {
            basic.clearScreen()
        }
    }
})
function callArrow () {
    degree = input.compassHeading()
    if (degree >= 337.5 || degree <= 22.5) {
        basic.showLeds(`
            . . # . .
            . # # # .
            # . # . #
            . . # . .
            . . # . .
            `)
    } else if (degree <= 67.5) {
        basic.showLeds(`
            # # # # #
            # # . . .
            # . # . .
            # . . # .
            # . . . #
            `)
    } else if (degree <= 112.5) {
        basic.showLeds(`
            . . # . .
            . # . . .
            # # # # #
            . # . . .
            . . # . .
            `)
    } else if (degree <= 157.5) {
        basic.showLeds(`
            # . . . #
            # . . # .
            # . # . .
            # # . . .
            # # # # #
            `)
    } else if (degree <= 202.5) {
        basic.showLeds(`
            . . # . .
            . . # . .
            # . # . #
            . # # # .
            . . # . .
            `)
    } else if (degree <= 247.5) {
        basic.showLeds(`
            # . . . #
            . # . . #
            . . # . #
            . . . # #
            # # # # #
            `)
    } else if (degree <= 292.5) {
        basic.showLeds(`
            . . # . .
            . . . # .
            # # # # #
            . . . # .
            . . # . .
            `)
    } else {
        basic.showLeds(`
            # # # # #
            . . . # #
            . . # . #
            . # . . #
            # . . . #
            `)
    }
}
let degree = 0
let printing = false
let ledOn = false
ledOn = false
printing = false
input.calibrateCompass()
basic.forever(function () {
    while (ledOn && !(printing)) {
        callArrow()
    }
})

File: compass.py
def on_button_pressed_a():
    global printing
    basic.clear_screen()
    printing = True
    basic.show_number(input.compass_heading())
    printing = False
input.on_button_pressed(Button.A, on_button_pressed_a)

def on_button_pressed_ab():
    callArrow()
    basic.pause(1000)
    basic.clear_screen()
input.on_button_pressed(Button.AB, on_button_pressed_ab)

def on_button_pressed_b():
    global ledOn
    if not (ledOn):
        ledOn = True
    else:
        ledOn = False
        if not (printing):
            basic.clear_screen()
input.on_button_pressed(Button.B, on_button_pressed_b)

def callArrow():
    global degree
    degree = input.compass_heading()
    if degree >= 337.5 or degree <= 22.5:
        basic.show_leds("""
            . . # . .
                        . # # # .
                        # . # . #
                        . . # . .
                        . . # . .
        """)
    elif degree <= 67.5:
        basic.show_leds("""
            # # # # #
                        # # . . .
                        # . # . .
                        # . . # .
                        # . . . #
        """)
    elif degree <= 112.5:
        basic.show_leds("""
            . . # . .
                        . # . . .
                        # # # # #
                        . # . . .
                        . . # . .
        """)
    elif degree <= 157.5:
        basic.show_leds("""
            # . . . #
                        # . . # .
                        # . # . .
                        # # . . .
                        # # # # #
        """)
    elif degree <= 202.5:
        basic.show_leds("""
            . . # . .
                        . . # . .
                        # . # . #
                        . # # # .
                        . . # . .
        """)
    elif degree <= 247.5:
        basic.show_leds("""
            # . . . #
                        . # . . #
                        . . # . #
                        . . . # #
                        # # # # #
        """)
    elif degree <= 292.5:
        basic.show_leds("""
            . . # . .
                        . . . # .
                        # # # # #
                        . . . # .
                        . . # . .
        """)
    else:
        basic.show_leds("""
            # # # # #
                        . . . # #
                        . . # . #
                        . # . . #
                        # . . . #
        """)
degree = 0
printing = False
ledOn = False
ledOn = False
printing = False
input.calibrate_compass()

def on_forever():
    while ledOn and not (printing):
        callArrow()
basic.forever(on_forever)
